### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by reefebba** [neomerx/json-api#229]
It would be very nice of this package if it has more beginner-friendly and all-covered documentation with examples.
However, Thank you, this package is very helpful

**Comment by neomerx** [comment]
@reefebba Hi, there is a [wiki](https://github.com/neomerx/json-api/wiki) which is intended to provide such information and [sample aplication](https://github.com/neomerx/json-api/tree/master/sample) that demonstrate how to work with the library. Also, there is a ton of [tests](https://github.com/neomerx/json-api/tree/master/tests/Encoder) that demonstrate every single aspect of the lib usage (mind you 100% test coverage).

If you haven't seen them, please have a look and drop a few lines with feedback. Is it enough? How easy it understandable for a new user? Do you see any changes that could make it better? Don't hesitate to send a PR or ask a question.

**Comment by JustSteveKing** [comment]
The examples and documentation provided show a very nice and easy way to use this package, but more indepth API docs could be useful to know what the constrains of the package are. I can kind of dig around and figure it out - but not all developers are going to be comfortable doing this

**Comment by nebur81** [comment]
I also believe that more documentation is necessary. If this framework had API docs I think it would be a great framework. But right now you lose a lot of time in guessing how certain things work

**Comment by neomerx** [comment]
@nebur81 Have you looked at Wiki? The API is actually quite simple. Don't hesitate to ask if anything is unclear or send a PR to the wiki to make it better.

**Comment by nebur81** [comment]
@neomerx  Yes, I have looked at the wiki. Don't get me wrong, I think it's a great framework, It's only as a comment says: maybe more indepth API docs could be useful. I'm a newbie with jsonapi 
Anyway I have a doubt, I make a test to encode an author (using some of your test files) and when I test it with Postman says that: **Content-Type →text/html; charset=UTF-8**
Content type should not be application/vnd.api+json?. 

My code to encode is: `echo $encoder->encodeData($author);`

I cand send an email, if you want and need it, to explain what I made or my code (is a very small example)

**Comment by neomerx** [comment]
@nebur81 Can you please open a new issue with your question? I'll remove our conversation from here to avoid duplicating.

**Comment by Doqnach** [comment]
Would love to see some documentation on how to handle a `POST`/`PUT`/`PATCH` with a (new) resource and one with a collection of (new) resources.

**Comment by neomerx** [comment]
@Doqnach Handling HTTP is out of the project's scope. You can use any framework of your choice to handle HTTP and use the library to encode your models/resources into JSON.

**Comment by Doqnach** [comment]
@neomerx I understand that since this is a framework agnostic library, that handling that is out of scope.

But I was expecting there to be some way to convert (incoming) JSON to models/resources, and not only the other way around? And do some validation of the payload coming in?
